File: backend/app/schemas/ai_generate.py
"""AI 生成 API 的 Pydantic schemas — re-exported from the router."""

from typing import Optional

from pydantic import BaseModel, Field


class AutoTagRequest(BaseModel):
    work_id: str


class AutoTagResponse(BaseModel):
    tags: list[str] = Field(default_factory=list)
    confidence: float = 0.0


class AutoDescriptionRequest(BaseModel):
    work_id: str


class AutoDescriptionResponse(BaseModel):
    description: str = ""


class ArticleDraftRequest(BaseModel):
    prompt: str
    tone: str = "professional"
    max_words: int = 2000


class ProductDescRequest(BaseModel):
    product_name: str
    materials: list[str] = Field(default_factory=list)
    techniques: list[str] = Field(default_factory=list)


class MusicDescRequest(BaseModel):
    title: str
    genre: str
    mood: str
    bpm: int


class ModerateRequest(BaseModel):
    text: str


class ModerateResponse(BaseModel):
    safe: bool = True
    categories: dict[str, bool] = Field(default_factory=dict)
    reason: str = ""


class AIConfigResponse(BaseModel):
    configured: bool
    provider: str = "openai"

File: frontend/src/api/ai_generate.ts
import client from './client'

export const aiGenerateApi = {
  autoTags: (workId: string) =>
    client.post<{ data: { tags: string[]; confidence: number } }>(
      '/ai/generate/tags',
      { work_id: workId },
    ),

  autoDescription: (workId: string) =>
    client.post<{ data: { description: string } }>(
      '/ai/generate/description',
      { work_id: workId },
    ),

  draftArticle: (prompt: string, tone: string = 'professional', maxWords: number = 2000) =>
    client.post<{ data: string }>('/ai/generate/article', {
      prompt,
      tone,
      max_words: maxWords,
    }),

  draftProductDesc: (
    productName: string,
    materials: string[],
    techniques: string[],
  ) =>
    client.post<{ data: string }>('/ai/generate/product-desc', {
      product_name: productName,
      materials,
      techniques,
    }),

  draftMusicDesc: (title: string, genre: string, mood: string, bpm: number) =>
    client.post<{ data: string }>('/ai/generate/music-desc', {
      title,
      genre,
      mood,
      bpm,
    }),

  moderateContent: (text: string) =>
    client.post<{
      data: { safe: boolean; categories: Record<string, boolean>; reason: string }
    }>('/ai/generate/moderate', { text }),

  getConfig: () =>
    client.get<{ data: { configured: boolean; provider: string } }>('/ai/generate/config'),
}
